zeebe/msgpack-value/src/main/java/io/camunda/zeebe/msgpack/property/BaseProperty.java

```diff
@@ -98,16 +103,48 @@ public void write(final MsgPackWriter writer) {
     valueToWrite.write(writer);
   }
 
-  public void writeJSON(final StringBuilder sb) {
+  public void writeJSON(final StringBuilder sb, final boolean maskSanitized) {
     key.writeJSON(sb);
     sb.append(":");
     if (hasValue()) {
-      resolveValue().writeJSON(sb);
+      if (maskSanitized && isSanitized) {
+        sb.append("\"***\"");
+      } else {
+        resolveValue().writeJSON(sb);
+      }
     } else {
       sb.append("\"NO VALID WRITEABLE VALUE\"");
     }
   }
 
+  public boolean isSanitized() {
+    return isSanitized;
+  }
+
+  /**
+   * If set to true, will be replaced by '***' when writing to console or logs, i.e. when called via
+   * {@link #toString()}, or {@link #writeJSON(StringBuilder, boolean)} with the second param set to
+   * true.
+   *
+   * <p>If false, disabled sanitization of this value in the cases above.
+   *
```

Comment:
:x: This comment doesnt make sense here

Reply:
Lol, it's a left over thanks to AI :smile: Initially there were some use cases, and I guess I removed them, but didn't update this :see_no_evil: